metadata-io/src/main/java/com/linkedin/metadata/graph/neo4j/Neo4jGraphService.java

```diff
@@ -179,8 +262,112 @@ private String generateLineageStatement(@Nonnull Urn entityUrn, @Nonnull Lineage
 
     final String allowedEntityTypes = String.join(" OR b:", graphFilters.getAllowedEntityTypes());
 
-    final String statementDirect = String.format(multiHopTemplateDirect, entityUrn, upstreamRel, maxHops, allowedEntityTypes, entityUrn);
-    final String statementIndirect = String.format(multiHopTemplateIndirect, entityUrn, dowStreamRel, maxHops, allowedEntityTypes, entityUrn);
+    final String multiHopMatchTemplateDirect = "MATCH p = shortestPath((a {urn: '%s'})-[r:%s*1..%d]->(b)) ";
```

Comment:
similar comment about a utils here - can we break this into smaller readable functions for each piece? everything below this in this method is a big chunk of code that's hard to decipher.